wandb/sdk/data_types/base_types/wb_value.py

```diff
@@ -24,16 +24,16 @@ def _server_accepts_client_ids() -> bool:
     # The latest SDK version that is < "0.11.0" was released on 2021/06/29.
     # AS OF NOW, 2024/11/06, we assume that all customer's server deployments accept
     # client IDs.
-    #
-    # If there are any users with issues on an older backend, customers can disable the
-    # setting `allow_offline_artifacts` to revert the SDK's behavior back to not
-    # using client IDs in offline mode.
-    if (
-        util._is_offline()
-        and wandb.run
-        and not wandb.run.settings.allow_offline_artifacts
-    ):
-        return False
+
+    if util._is_offline():
+        # If there are any users with issues on an older backend, customers can disable the
+        # setting `allow_offline_artifacts` to revert the SDK's behavior back to not
+        # using client IDs in offline mode.
+        if wandb.run and not wandb.run.settings.allow_offline_artifacts:
+            return False
+        # Assume client IDs are accepted
+        else:
+            return True
```

Comment:
i think it is relatively safe assumption, but it would break old server users, if you want to go ahead and add this change i would suggest updating the changelog, currently it is not very informative

Reply:
oh i guess they can disable it by using `allow_offline_artifacts` as false, so i think we have a solution for these users.

Reply:
still would be useful to add in changelog that the assumption is that the server accepts offline artifacts (corner case is if `wandb.run` wasn't defined we blindly assume that the server accepts these and it would require the user to modify their script to ad the relevant setting.